libraries/AC_Fence/AC_PolyFence_loader.cpp

```diff
@@ -180,30 +183,57 @@ bool AC_PolyFence_loader::load_point_from_eeprom(uint16_t i, Vector2l& point)
 
 bool AC_PolyFence_loader::breached() const
 {
+    bool is_breached = false;
+    bool have_position = false;
+    #if POLYFENCE_BREACH_TIMINGS
+    void *istate = hal.scheduler->disable_interrupts_save();
+    #endif
+    hal.util->perf_begin(_perf_BreachCalc);
+    uint64_t start_time = AP_HAL::micros64();
     // check if vehicle is outside the polygon fence
-    Vector2f position;
-    if (!AP::ahrs().get_relative_position_NE_origin(position)) {
-        // we have no idea where we are; can't breach the fence
-        return false;
-    }
-
-    position = position * 100.0f;  // m to cm
-    return breached(position);
+    // this is done with the lat/lon coordinates of the vehicle
+    struct Location loc;
+    if (AP::ahrs().get_position(loc)) {
+        have_position = true;
+        Vector2l position;
+        position.x = loc.lat;
+        position.y = loc.lng;
+        is_breached = breached(position);
+    }
+
+    uint64_t end_time = AP_HAL::micros64();
+    hal.util->perf_end(_perf_BreachCalc);
+
+    AP::logger().Write("PLYF", 
+                        "TimeUS,Diff,Brchd,Pos",
+                        "----",
+                        "----",
+                        "QQBb",
+                        AP_HAL::micros64(),
+                        (end_time - start_time),
+                        is_breached,
+                        have_position);
+
+    #if POLYFENCE_BREACH_TIMINGS
+    hal.scheduler->restore_interrupts(istate);
+    #endif
+
+    return is_breached;
 }
 
 bool AC_PolyFence_loader::breached(const Location& loc) const
 {
-    Vector2f posNE;
-    if (!loc.get_vector_xy_from_origin_NE(posNE)) {
-        // not breached if we don't now where we are
-        return false;
-    }
-    return breached(posNE);
+    // Check if a supplied location breaches fence zones
+    // Does not need a home location in order to determine breach status
+    Vector2l pos;
+    pos.x = loc.lat;
+    pos.y = loc.lng;
+    return breached(pos);
 }
 
-// check if a position (expressed as offsets in cm from the EKF origin) is within the boundary
+// check if a position (expressed as lat/lng) is within the boundary
 //   returns true if location is outside the boundary
-bool AC_PolyFence_loader::breached(const Vector2f& pos_cm) const
+bool AC_PolyFence_loader::breached(const Vector2l& pos) const
```

Comment:
I'm not a fan of introducing another way to express a position/location.  We already have Location and Vector3f (which is always cm from the ekf origin).  I think adding another Vector2l is confusing and we might as well keep the interface using Locations shouldn't we?

Reply:
We could condense this (after the update made to address removing of the param) to two functions. We still need to pull the `Vector2l` information from the location in order to call into the Math library.
* `breached()` 
* `breached(const Location* loc)`

I can update this.